flocker/node/agents/blockdevice.py

```diff
@@ -591,6 +622,8 @@ def run(self, deployer):
         return succeed(None)
 
 
+# Get rid of this in favor of calculating each individual operation in
+# BlockDeviceDeployer.calculate_changes.  FLOC-1771
```

Comment:
So eventually, calculate_state_changes will detect eg a block device for a  dataset that isn't formatted and yield a CreateFilesystem change ONLY.
Then in the next convergence loop it'll detect block device for a dataaset that isn't mounted and yield `MountBlockDevice`.

But then won't we need more block device states than `manifest` /  `non-manifest` in the Deployment state? 
...hmm i suppose not 
 * `discover_state` will have to only report mounted filesystems as `manifest` (perhaps it already does)
 * `calculate_changes` then examines the exact state of all non-manifest datasets and yields the correct next state change.